libraries/NonFungibleToken/contracts/non-fungible.aes

```diff
@@ -1,110 +1,75 @@
+    require(is_approved_or_owner(Call.caller, token_id), "Caller isn't approved or owner")
     
-    clearApproval(from, tokenId)
-    removeTokenFrom(from, tokenId)
-    addTokenTo(to, tokenId)
-
-    true
-
-  public stateful function setApprovalForAll(to : address, approved : bool) : bool =
-    ensureAllowed(Call.caller)
-    put(state{operatorApprovals[Call.caller][to] = approved})
-    true
-  
-  private function checkTokenExists(tokenId : int) : bool =
-    switch(Map.lookup(tokenId, state.tokenOwner))
-      None    => false
-      Some(x) => true
-
-  private function ensureAllowed(key : address) =
-    switch(Map.lookup(key, state.operatorApprovals))
-      None => put(state{operatorApprovals[key] = {}})
-      Some(_) => ()
-
-  private stateful function addTokenTo(to : address, tokenId : int) = 
-    put(state{tokenOwner = state.tokenOwner{[tokenId] = to}})
-    put(state{ownedTokensCount[to] = add(lookupByAddress(to, state.ownedTokensCount, 0), 1)})
-
-  private stateful function removeTokenFrom(from : address, tokenId : int) =
-    put(state{tokenOwner = state.tokenOwner{[tokenId] = #0}})
-    put(state{ownedTokensCount[from] = sub(lookupByAddress(from, state.ownedTokensCount, 0), 1)})
-
-  private stateful function clearApproval(from : address, tokenId : int) : bool = 
-    put(state{tokenApprovals = state.tokenApprovals{[tokenId] = #0}})
-    true
-
-  private function lookupByAddress(k : address, m, v) =
-    switch(Map.lookup(k, m))
-      None    => v
-      Some(x) => x
-
-  private function lookupById(k : int, m, v) =
-    switch(Map.lookup(k, m))
-      None    => v
-      Some(x) => x
-
-  private function add(_a : int, _b : int) : int =
-    let c : int = _a + _b
-    require(c >= _a, "Error")
-    c
-
-  private function sub(_a : int, _b : int) : int =
-    require(_b =< _a, "Error")
-    _a - _b
-
-  //todo use native abort!!
-  private function abort(err) = abort(err)
-
-  private function require(b : bool, err : string) =
-    if(!b) 
-      abort(err)
-
-  private function onlyOwner(owner : address) =
-    if(owner != state.owner) 
-      abort("Only owner can mint!")
-
-  private function onlyTokenOwner(owner : address, tokenId : int) =
-    if(owner != lookupById(tokenId, state.tokenOwner, #0)) 
-      abort("Only token owner can transfer!")
-
-  private function isApprovedOrOwner(spender : address, tokenId : int) : bool = 
-    let owner : address = ownerOf(tokenId)
-    getApproved(tokenId) == spender || owner == spender || isApprovedForAll(owner, spender)
\ No newline at end of file
+    clear_approval(from, token_id)
+    remove_token_from(from, token_id)
+    add_token_to(to, token_id)
+
+  public stateful function set_approval_for_all(to: address, approved: bool) =
+    put(state{m_operator_approvals[Call.caller = {}][to] = approved})
+
+  private stateful function add_token_to(to: address, token_id: int) = 
+    put(state{m_tokenOwner = state.m_tokenOwner{[token_id] = to}})
```

Comment:
`put(state{m_tokenOwner = state.m_tokenOwner{[token_id] = to}})`

to 

`put(state{m_token_owner = state.m_token_owner{[token_id] = to}})`